llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp

```diff
@@ -1556,6 +1556,9 @@ static bool processDbgDeclare(FunctionLoweringInfo &FuncInfo,
   if (processIfEntryValueDbgDeclare(FuncInfo, Address, Expr, Var, DbgLoc))
     return true;
 
+  if (!Address->getType()->isPointerTy())
+    return false;
```

Comment:
This part is untested (and needs globalisel equivalent?)

Reply:
Restored some tests I originally adjusted.

I just looked at IRTranslator, and it doesn't seem to have the offset stripping code that caused this to assert.